### PR TITLE
Fix error handeling for dataframe for str.cat #5621

### PR DESCRIPTION
**Pull request by sephib**
- [X ] Tests added / passed
- [X ] Passes `black dask` / `flake8 dask`

Update str.cat function to handle dataframe input

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -298,3 +298,17 @@
...
+                                  'http://pandas.pydata.org/pandas-docs/stable/objects.inv'),
+                       'numpy': ('https://docs.scipy.org/doc/numpy/',
+                                 'https://docs.scipy.org/doc/numpy/objects.inv')}
+
+#  --Options for sphinx extensions -----------------------------------------------
+
+intersphinx_mapping = {'pandas': ('http://pandas.pydata.org/pandas-docs/stable/',
+                                  'http://pandas.pydata.org/pandas-docs/stable/objects.inv'),
+                       'numpy': ('https://docs.scipy.org/doc/numpy/',
+                                 'https://docs.scipy.org/doc/numpy/objects.inv')}
```
Comment (jacobtomlinson) [inline docs/source/conf.py]: Looks like this section has been duplicated here.

Comment (sephib) [reply docs/source/conf.py]: Some old files have joined this PR.
I'm going to remove them and resubmit with the changes of only relevant for str.cat

Comment (jacobtomlinson) [reply docs/source/conf.py]: Ok sure. Is there a PR for the intersphinx stuff somewhere?

Comment (jacobtomlinson) [reply docs/source/conf.py]: Ah found it #3050. Yeah I assume this is a "squash and merge" issue.

**Pull request closed without merging**